bin/improver-nbhood-land-and-sea

```diff
@@ -0,0 +1,236 @@
+                       'valid. The radii are in metres whilst the lead time '
+                       'has units of hours. The radii and lead times are '
+                       'expected as individual comma-separated lists with '
+                       'the list of radii given first followed by a list of '
+                       'lead times to indicate at what lead time each radii '
+                       'should be used. For example: 10000,12000,14000 1,2,3 '
+                       'where a lead time of 1 hour uses a radius of 10000m, '
+                       'a lead time of 2 hours uses a radius of 12000m, etc.')
+
+    parser.add_argument('--ens_factor', metavar='ENS_FACTOR', type=float,
+                        default=1.0,
+                        help='The factor with which to adjust the '
+                        'neighbourhood size for more than one '
+                        'ensemble member. If ens_factor = 1.0 this '
+                        'essentially conserves ensemble members if '
+                        'every grid square is considered to be the '
+                        'equivalent of an ensemble member.'
+                        'Optional, defaults to 1.0.')
+    parser.add_argument('--sum_or_fraction', default="fraction",
+                        choices=["sum", "fraction"],
+                        help='The neighbourhood output can either be in the '
+                             'form of a sum of the neighbourhood, or a '
+                             'fraction calculated by dividing the sum of the '
+                             'neighbourhood by the neighbourhood area. '
+                             '"fraction" is the default option.')
+    parser.add_argument('--no_clip', action='store_true', default=False,
+                        help="By default the results of neighbourhooding will "
+                             "be clipped to the extremes in the input data. "
+                             "If '--no_clip' is used then this clipping is "
+                             "turned off. "
+                             "Once the neighbourhood code is fixed so it "
+                             "doesn't produce values above or below the "
+                             "extremes in the input data this option can be "
+                             "removed.")
+    parser.add_argument('--intermediate_filepaths', default=None, nargs=2,
+                        help='If provided the results after neighbourhooding, '
+                        'but before recombining land and sea are saved to the '
+                        'two given filepaths; the first path for the land '
+                        'output, the second path for the sea output.')
+
+    args = parser.parse_args()
+
+    cube = load_cube(args.input_filepath)
+    mask = load_cube(args.input_mask_filepath)
+    masking_coordinate = None
+
+    if any(['topographic_zone' in coord.name()
```

Comment:
Might it be easier to put a switch in as an option to the CLI to tell you whether you want to do topographic masking or not. This has also been hard-coded to only support topographic_zones, in principle the iterate-with-mask CLI could be anything, but topographic zones is the only use case we have at the moment, so maybe this hard-coding is ok.

Reply:
There is a bit more cube manipulation in here than is ideal, but I think that's ok for. We need to revisit how all the neighbourhood code is structured anyway.